src/tools/execution-results.h

```diff
@@ -151,9 +151,14 @@ struct ExecutionResults {
   bool areEqual(Literal a, Literal b) {
     // We allow nulls to have different types (as they compare equal regardless)
     // but anything else must have an identical type.
-    if (a.type != b.type && !(a.isNull() && b.isNull())) {
-      std::cout << "types not identical! " << a << " != " << b << '\n';
-      return false;
+    // We cannot do this in nominal typing, however, as different modules will
+    // have different types in general. We could perhaps compare them
+    // structurally, but that would not be right either.
```

Comment:
Comparing structures one pair of types at a time wouldn't work, but we can at least have a TODO about checking for structural equivalence between the definition graphs.

Reply:
Sounds good, added TODO.